superset-frontend/src/explore/components/DatasourcePanel/index.tsx

```diff
@@ -190,6 +217,35 @@ export default function DataSourcePanel({
     [_columns],
   );
 
+  const placeholderSlDataset = {
+    sl_table: [],
+    query: [],
+    saved_query: [],
+  };
+
+  // eslint-disable-next-line no-param-reassign
+  datasource.sl_dataset = placeholderSlDataset;
+
+  const getDefaultDatasetName = () =>
+    `${datasource?.sl_dataset?.query.tab} ${moment().format(
```

Comment:
can you just reference the datasource directly? `datasource?.query.tab`

Reply:
There's currently no `query` on DatasourceMeta, should I add it?